nautobot_golden_config/models.py

```diff
@@ -862,33 +862,41 @@ def get_absolute_url(self):
 class ConfigPlan(PrimaryModel):  # pylint: disable=too-many-ancestors
     """ConfigPlan for Golden Configuration Plan Model definition."""
 
-    plan_type = models.CharField(max_length=20, choices=ConfigPlanTypeChoice)
+    plan_type = models.CharField(max_length=20, choices=ConfigPlanTypeChoice, verbose_name="Plan Type")
     device = models.ForeignKey(
         to="dcim.Device",
         on_delete=models.CASCADE,
         related_name="config_plan",
     )
     config_set = models.TextField(help_text="Configuration set to be applied to device.")
-    feature = models.ForeignKey(
-        to="ComplianceFeature",
-        on_delete=models.CASCADE,
+    feature = models.ManyToManyField(
```

Comment:
as above

Reply:
Other consideration : 
- config plan hash

I believe the hash should be also available even before generating a plan and attached to `missing , intended , remediation` in the compliance. This should allow to create config plans that are the same, giving more and earlier visibility into this process.

If we allow for many to many here , we will have less accurate config plan hashes , thus less possibilities for aggregation.

Reply:
One more thing to be considered :

- multiple features per configPlan will not allow to directly correlate Compliance Results (like missing, intended and extra) with ConfigSet inside ConfigPlan.

This can be important for cases where automated process should validate if there was a change in the "source information" that we used to generate ConfigPlan. As the ConfigPlan introduces approval stage, we expect days/weeks in the process before it gets approved. This time should be considered and it should be understood that there could be the data change or configuration change that the plan is not aware of. Because of that, natural and sane would be to introduce pre-change validation.